app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -215,6 +215,12 @@ class HomeFragment : Fragment() {
             sessionControlInteractor,
             homeViewModel
         )
+
+        // This has to be called separately from the consumeFrom block since the coroutine doesn't
+        // allow our UI to be updated right away and delays our start up. The init block allows us to
+        // force our UI to render with the data available in our store at fragment creation time.
+        sessionControlView?.update(homeFragmentStore.state)
```

Comment:
**Big issue**: I don't love that we're "setting top sites twice" but only dispatching the change once but _it seems_ innocuous on P2. However, on my GS5, the whole homescreen flashes, presumably because it relays out the RV after this line: https://searchfox.org/mozilla-mobile/rev/d81a3c3fce58d3a6c3ebd3257ad1566c9bd462ef/fenix/app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt#147 The current behavior does this too but it doesn't look as bad because top sites gets added instead of just replaced.

I'm not sure what we should do about that... Let's discuss

Reply:
Interesting. I don't see it on my G5. We could do a quick "if this is the first call from the consumeFrom, ignore"

Reply:
The problem is fixed if we remove that line. The line was added in https://github.com/mozilla-mobile/fenix/pull/5591 and was trying to fix https://github.com/mozilla-mobile/fenix/issues/2874, https://github.com/mozilla-mobile/fenix/issues/5131, and https://github.com/mozilla-mobile/fenix/issues/5375. The first two relate to problems with tabs updating on the homescreen – no longer a problem – and I'm unable to reproduce the 3rd issue (maybe also tabs related)? I tried reproducing similar issues with collections:
- open homescreen, open tabs tray, click 3-dot to add to collection, see homescreen
- open history/bookmarks, add collection, open homescreen

But I can't reproduce. I think this is safe to remove but we should check with ekager.